RxSwift/Observable+Concurrency.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  Observable+Concurrency.swift
+//  RxSwift
+//
+//  Created by Shai Mishali on 22/09/2021.
+//  Copyright © 2021 Krunoslav Zaher. All rights reserved.
+//
+
+#if swift(>=5.5) && canImport(_Concurrency)
+import Foundation
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public extension ObservableConvertibleType {
+    /// Allows iterating over the values of an Observable
+    /// asynchronously via Swift's concurrency features (`async/await`)
+    ///
+    /// A sample usage would look like so:
+    ///
+    /// ```swift
+    /// do {
+    ///     for try await value in observable.values {
+    ///         // Handle emitted values
+    ///     }
+    /// } catch {
+    ///     // Handle error
+    /// }
+    /// ```
+    var values: AsyncThrowingStream<Element, Error> {
+        AsyncThrowingStream<Element, Error> { continuation in
+            _ = asObservable().subscribe(
+                onNext: { value in continuation.yield(value) },
+                onError: { error in continuation.finish(throwing: error) },
+                onCompleted: { continuation.finish() },
+                onDisposed: { continuation.onTermination?(.cancelled) }
+            )
+        }
+    }
```

Comment:
Returned `Disposable` is not used, so cancellation is probably not working?

Reply:
Maybe this code might be useful.
https://twitter.com/inamiy/status/1439431498457759745

Reply:
Generally speaking `Disposable`s are deallocated when leaving the closest scope so this will only be an issue when there's explicit cancellation of the task or its parent. Anyways, I don't think `withExtendedLifetime` will be needed :)

Great thoughts anyways, I'll add that. Thank you 🙏

Reply:
Yes, above twitter link is about Combine, not RxSwift 🚜
But the cancellation approach will still be similar where `onTermination` handler registration becomes necessary.

In RxSwift, it will require manual `disposable.dispose()` call rather than Combine's auto-cancellation on deinit.

For example:

```swift
continuation.onTermination = { @Sendable termination in
    switch termination {
    case .cancelled:
        disposable.dispose()
    case .finished:
        disposable.dispose() // probably won't need, but just in case...
    }
}
```

I need to elaborate more on `@Sendable`, but I guess this attribute will probably be needed too.

Reply:
Yup, makes sense. I'll take a look, thank you!